pkg/util/detector/detector.go

```diff
@@ -216,12 +225,27 @@ func (d *ResourceDetector) EventFilter(obj interface{}) bool {
 		return false
 	}
 
-	if clusterWideKey.Group == clusterv1alpha1.GroupName ||
-		clusterWideKey.Group == policyv1alpha1.GroupName ||
-		clusterWideKey.Group == workv1alpha1.GroupName {
-		return false
+	if d.ResourceConfig != nil {
+		gv := schema.GroupVersion{Group: clusterWideKey.Group,
+			Version: clusterWideKey.Version}
+		gvk := schema.GroupVersionKind{
+			Group:   clusterWideKey.Group,
+			Version: clusterWideKey.Version,
+			Kind:    clusterWideKey.Kind,
+		}
+		if d.ResourceConfig.GroupDisabled(clusterWideKey.Group) {
+			klog.V(4).Infof("Skip event for %s", clusterWideKey)
+			return false
+		}
+		if d.ResourceConfig.GroupVersionDisabled(gv) {
+			klog.V(4).Infof("Skip event for %s", clusterWideKey)
+			return false
+		}
+		if d.ResourceConfig.GroupVersionKindDisabled(gvk) {
+			klog.V(4).Infof("Skip event for %s", clusterWideKey)
+			return false
+		}
```

Comment:
I guess we didn't make an agreement about this flag.

The flag used to specify `extra` APIs should be ignored, not used to `replace` the defaults.

Reply:
yes, I moved these default into 
```
	// disable karmada group by default
	r.DisableGroup(clusterv1alpha1.GroupVersion.Group)
	r.DisableGroup(policyv1alpha1.GroupVersion.Group)
	r.DisableGroup(workv1alpha1.GroupVersion.Group)
```